src/launcher.rs

```diff
@@ -396,11 +406,42 @@ impl Launcher {
         Ok(())
     }
 
+    /// Runs the shutdown script if it exists and exits the node-launcher process
+    /// with the exit code returned by the script, otherwise returns 0.
```

Comment:
In it's current form it can also return `1`, but see the comment below.

Reply:
I think the inaccuracy in the comment is `exits the node-launcher process with the exit code returned by the script`, not `otherwise returns 0`. The case where we return `1` here is when the script is killed by a signal, so that means the script ran, but unsuccessfully. By definition, processes killed by signals do not have exit codes, so we need some convention here to establish that a signal killed the process.